### PR TITLE
Make Treating Doctor Field Optional for Domiciliary Care Patients #8304

### LINKED ISSUE #8304: Treating Doctor field should not be mandatory for Domiciliary Care patients
**Describe the problem**
Currently treating doctor field is mandatory to all type of patients for filing consultation. 

However in Domicilary care scenario there wont be a doctor assigned at first to visit the patient at home and update consultation details. usually nurses will  update these details. So the mandatory treating doctor field is restricting them to update patient details smoothly. 

1. create a patient file
2. select decision after consultation as Domiciliary Care
3. Scroll down treating doctor field

**Suggested solution**
Treating doctor field shouldn't be mandatory for Domiciliary care patients. This is, consultation forms where "Domiciliary care" is selected under decision after consultation, Treating doctor field shouldn't be mandatory.

### PR DESCRIPTION
**Pull request by JOSHIK27**
## Required Backends

- https://github.com/coronasafe/care/pull/2382

## Proposed Changes

- Fixes #8304 

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/ConsultationForm.tsx [hunk 1]
```diff
@@ -1444,7 +1448,9 @@ export const ConsultationForm = ({ facilityId, patientId, id }: Props) => {
                           name={"treating_physician"}
                           label={t("treating_doctor")}
                           placeholder="Attending Doctors Name and Designation"
-                          required
+                          required={
+                            state.form.suggestion === "DC" ? false : true
+                          }
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationForm.tsx]:
```suggestion
                          required={state.form.suggestion !== "DC"}
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by rithviknishad** [comment]
@JOSHIK27 Could you also update the PR's title to be something more meaningful 😄

**Comment by JOSHIK27** [comment]
> @JOSHIK27 Could you also update the PR's title to be something more meaningful 😄

done @rithviknishad

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@JOSHIK27 
image

its throwing a 500 error in the frontend 

CC: @rithviknishad

**Comment by rithviknishad** [comment]
@nihal467 backend changes required: https://github.com/coronasafe/care/pull/2382

Could you test with that?

**Comment by rithviknishad** [comment]
@JOSHIK27 would you be making a new PR for this?

**Comment by JOSHIK27** [comment]
> @JOSHIK27 would you be making a new PR for this?

Yes. Definitely

**Pull request closed without merging**